playhouse/fields.py

```diff
@@ -26,12 +28,43 @@
 except ImportError:
     AES = Random = None
 
+try:
+    from bcrypt import hashpw, gensalt
+except ImportError:
+    hashpw, gensalt = None, None
+
 from peewee import *
 from peewee import binary_construct
 from peewee import Field
 from peewee import FieldDescriptor
 from peewee import SelectQuery
 
+
+class PasswordHash(bytes):
+            def check_password(self, password):
+                password = password.encode('utf-8')
+                return hashpw(password, self) == self
+
+
+class PasswordField(TextField):
+    def __init__(self, iterations=12, *args, **kwargs):
+        if None in (hashpw, gensalt):
+            raise ValueError('Missing library required for PasswordField: bcrypt')
+        self.bcrypt_iterations = iterations
+        self.raw_password = None
+        super(PasswordField, self).__init__(*args, **kwargs)
+
+    def db_value(self, value):
+        """Convert the python value for storage in the database."""
+        value = value.encode('utf-8')
+        salt = gensalt(self.bcrypt_iterations)
+        return value if value is None else hashpw(value, salt)
+
+    def python_value(self, value):
+        """Convert the database value to a pythonic value."""
+        # FixMe: how do i get this to run before saving to the DB?
```

Comment:
It'd be really cool to do this when i create the password instead of when I get it from the DB.
But i have no idea how to go about that.